src/goto-analyzer/goto_analyzer_parse_options.cpp

```diff
@@ -229,8 +229,7 @@ void goto_analyzer_parse_optionst::get_command_line_options(optionst &options)
   }
   else if(cmdline.isset("intervals") || cmdline.isset("non-null"))
   {
-    // For backwards compatibility either of these on their own means show
```

Comment:
The key here is this part.  ```--intervals``` and ```--non-null``` on their own are ```--show``` for backwards compatibility.  If we can break that then the command handling can be significantly simplified.